### PR TITLE
[JENKINS-52024] Fix annotation processing to support compiling Core under Java 10

### PR DESCRIPTION
**Pull request by svanoort**
See [JENKINS-52024](https://issues.jenkins-ci.org/browse/JENKINS-52024).

Pulls in:

* https://github.com/jglick/sezpoz/pull/13
* https://github.com/kohsuke/metainf-services/pull/15

### Proposed changelog entries

* Java 10 Support: Update SezPoz and metainf-services plugins to fix Core module compile-time errors with JDK10 due to annotation processing failures

### Submitter checklist

- [x] JIRA issue is well described
- [x] Changelog entry appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
      * Use the `Internal: ` prefix if the change has no user-visible impact (API, test frameworks, etc.)
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

Like #3515 but applied to the **right** base branch.

### CODE REVIEW

**Comment by svanoort** [comment]
@dwnusbaum

File: core/pom.xml [hunk 1]
```diff
@@ -570,8 +570,7 @@ THE SOFTWARE.
     <dependency>
       <groupId>org.kohsuke.metainf-services</groupId>
       <artifactId>metainf-services</artifactId>
-      <version>1.4</version>
-      <scope>provided</scope>
```
Comment (dwnusbaum) [inline core/pom.xml]: Any reason for the provided scope earlier? Maybe related to https://github.com/jenkinsci/core-annotation-processors?

**Review by dwnusbaum — APPROVED** [review]
I am still getting `NoSuchFileException: /Users/dnusbaum/Projects/jenkins/cli/target/classes/META-INF/annotations/org.kohsuke.accmod.Restricted` for the CLI modle (maybe an issue with my local environment?), but this definitely fixes compilation for core!

**Comment by jglick** [comment]
> I am still getting `NoSuchFileException: /Users/dnusbaum/Projects/jenkins/cli/target/classes/META-INF/annotations/org.kohsuke.accmod.Restricted`

That is actually a different library, forked from SezPoz code IIUC. That error should have been fixed by https://github.com/jenkinsci/lib-annotation-indexer/pull/2, in `annotation-indexer 1.10`, long since picked up in core, so…?

**Comment by dwnusbaum** [comment]
@jglick 

> That error should have been fixed by jenkinsci/lib-annotation-indexer#2, in annotation-indexer 1.10, long since picked up in core, so…?

But not yet picked up by the CLI module, which is still using 1.4. I think @svanoort is planning to add an explicit dependency as part of this PR, but I will follow up with a bump in access-modifier-annotations.

**Review by oleg-nenashev — APPROVED** [review]
Looks good now, will be better with accmod

**Comment by svanoort** [comment]
Latest update fixes CLI -- if I build with:

`JAVA_HOME=$Java10InstallDirectory mvn clean install -Dmaven.test.skip -Dfindbugs.skip` it will not get any errors. 

Note though that I haven't verified that the final result is completely usable -- there's some grumbling in the build still (it looks like it might be during the WAR, but it floods the console).

File: core/pom.xml (hunk 1 above)
Comment (svanoort) [reply core/pom.xml]: @dwnusbaum Core-annotation-processors can be patched -- it has stale dependencies too.  I have a fast patch for that too, but need to clean it up a tad.

**Review by dwnusbaum — APPROVED** [review]

**Comment by oleg-nenashev** [comment]
@svanoort @dwnusbaum Should we enable the build on JDK10 for this branch?

**Comment by dwnusbaum** [comment]
@oleg-nenashev I say go for it, so we can settle whether there is an issue compiling the "Tests for Jenkins core" module or we have bad local environments :)

**Pull request merged**